src/formatDate.js

```diff
@@ -9,6 +9,81 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+    const separator = fromFormat[fromFormat.length - 1];
+    const currentCharacter = fromCharacter + element.length + separator.length;
```

Comment:
The `currentCharacter` calculation includes the length of the separator, which might not be correct if the `element` is the last in the `fromFormat` array. You should only add the separator's length if there is another date component following the current one.